iina/JavasctiptDevTool.swift

```diff
@@ -0,0 +1,506 @@
+          Button(action: { executePrompt(printGlobalObject: true) }) {
+            Image(systemName: "point.3.filled.connected.trianglepath.dotted")
+              .renderingMode(.template)
+              .foregroundColor(.secondary)
+          }
+          .buttonStyle(PlainButtonStyle())
+          .help("Print global objects")
+          Spacer()
+          Text("Cmd+Return to run code")
+            .font(.footnote)
+            .foregroundColor(.secondary)
+        }
+        .padding([.top, .horizontal], 8)
+        // Editor
+        CommandEditor(text: $input)
+          .frame(maxHeight: 100)
+          .padding(.all, 8)
+
+      }
+    }
+    .listStyle(.plain)
+  }
+
+  private func executePrompt(printGlobalObject: Bool = false) {
+    let source = printGlobalObject ? "$global" : input.trimmingCharacters(in: .whitespacesAndNewlines)
+    if source.isEmpty {
+      return
+    }
+    events.addPrompt(source)
+    events.postChange()
+
+    let jsResult: JSEvent.Result
+
+    let result = source == "$global" ?
+    jsContext.globalObject :
+    jsContext.evaluateScript(source)
+
+    if let result = result {
+      if result.isNumber {
+        jsResult = .number(result.toNumber())
+      } else if result.isString {
+        jsResult = .string(result.toString())
+      } else if result.isBoolean {
+        jsResult = .boolean(result.toBool())
+      } else if result.isNull {
+        jsResult = .null
+      } else if result.isUndefined {
+        jsResult = .undefined
+      } else if result.isArray {
+        let array = result.toArray()!
+        jsResult = .array(
+          "Array (\(array.count))", array.enumerated().map { ("\($0.offset)", "\($0.element)") } )
+      } else {
+        let object = result.toObject()!
+        if let dict = object as? [String: Any] {
+          jsResult = .object(
+            result.toString() ?? "", dict.keys.map { ($0, "\(dict[$0] ?? "nil")") })
+        } else {
+          jsResult = .opaqueObject(result.toString())
+        }
+      }
+      events.addResult(jsResult)
+      events.postChange()
+    }
+
+    if !printGlobalObject {
+      input = ""
+    }
+  }
+}
+
+@available(macOS 12.0, *)
+struct JavasctiptDevTool_Previews: PreviewProvider {
+  static var previews: some View {
+    JavasctiptDevTool(jsContext: JSContext())
+  }
+}
+
+
+// MARK: - Views
+
+@available(macOS 12.0, *)
+struct CommandEditor: NSViewRepresentable {
+  @Binding var text: String
+
+  init(text: Binding<String>) {
+    _text = text
+  }
+
+  func makeCoordinator() -> Coordinator {
+    Coordinator(self)
+  }
+
+  func makeNSView(context: Context) -> NSView {
+    let view = TextView()
+    view.delegate = context.coordinator
+    view.parent = self
+    return view
+  }
+
+  func updateNSView(_ nsView: NSView, context: Context) {
+    let textView = nsView as! NSTextView
+    textView.string = text
+    textView.font = .monospacedSystemFont(ofSize: NSFont.systemFontSize(for: .regular), weight: .regular)
```

Comment:
Worth sharing this font across SwiftUI and AppKit?

Reply:
This font is only used here, so I'm not sure what you mean…